Upstream-Status: Pending

Fix following compiler warining/errors with gcc 4.6.0.
Nitin A Kamble <nitin.a.kamble@intel.com> 2011/05/09

| kexec/kexec.c: In function 'my_exec':
| kexec/kexec.c:818:6: warning: variable 'result' set but not used [-Wunused-but-set-variable]

| ccache i586-poky-linux-gcc -march=i586 --sysroot=/build_disk/poky_build/build_gcc_4.6.0/tmp/sysroots/qemux86  --no-undefined -nostartfiles -nostdlib -nodefaultlibs -e purgatory_start -r -o purgatory/purgatory.ro purgatory/purgatory.o purgatory/printf.o purgatory/string.o purgatory/arch/i386/entry32-16.o purgatory/arch/i386/entry32-16-debug.o purgatory/arch/i386/entry32.o purgatory/arch/i386/setup-x86.o purgatory/arch/i386/stack.o purgatory/arch/i386/compat_x86_64.o purgatory/arch/i386/purgatory-x86.o purgatory/arch/i386/console-x86.o purgatory/arch/i386/vga.o purgatory/arch/i386/pic.o purgatory/arch/i386/crashdump_backup.o purgatory/sha256.o
| i586-poky-linux-gcc: error: unrecognized option '--no-undefined'
| make: *** [purgatory/purgatory.ro] Error 1
| ERROR: oe_runmake failed


Index: kexec-tools-2.0.2/kexec/kexec.c
===================================================================
--- kexec-tools-2.0.2.orig/kexec/kexec.c
+++ kexec-tools-2.0.2/kexec/kexec.c
@@ -815,7 +815,7 @@ static int my_shutdown(void)
  */
 static int my_exec(void)
 {
-	int result;
+	int __attribute__((__unused__)) result;
 
 	result = kexec_reboot();
 	/* I have failed if I make it here */
Index: kexec-tools-2.0.2/purgatory/Makefile
===================================================================
--- kexec-tools-2.0.2.orig/purgatory/Makefile
+++ kexec-tools-2.0.2/purgatory/Makefile
@@ -56,7 +56,7 @@ $(PURGATORY): CPPFLAGS=$($(ARCH)_PURGATO
 			-I$(srcdir)/include \
 			-I$(shell $(CC) -print-file-name=include)
 $(PURGATORY): LDFLAGS=$($(ARCH)_PURGATORY_EXTRA_CFLAGS)\
-			--no-undefined -nostartfiles -nostdlib -nodefaultlibs \
+			-Wl,--no-undefined -nostartfiles -nostdlib -nodefaultlibs \
 			-e purgatory_start -r
 
 $(PURGATORY): $(PURGATORY_OBJS)
